core/src/main/java/com/novoda/noplayer/mediaplayer/forwarder/CompletionInfoForwarder.java

```diff
@@ -0,0 +1,24 @@
+package com.novoda.noplayer.mediaplayer.forwarder;
+
+import android.media.MediaPlayer;
+
+import com.novoda.noplayer.listeners.InfoListeners;
+
+import java.util.HashMap;
+
+class CompletionInfoForwarder implements MediaPlayer.OnCompletionListener {
+
+    private final InfoListeners infoListeners;
+
+    public CompletionInfoForwarder(InfoListeners infoListeners) {
+        this.infoListeners = infoListeners;
+    }
+
+    @Override
+    public void onCompletion(MediaPlayer mp) {
+        HashMap<String, String> keyValuePairs = new HashMap<>();
+        keyValuePairs.put("mp", String.valueOf(mp));
+
+        infoListeners.onNewInfo("onCompletion", keyValuePairs);
```

Comment:
The problem with doing this is that you would need to have a new `onNewInfo` for each interface in exoplayer and media player that you implement. Which might be overkill when the client will just output these to a Log.